catalyst/metrics/avg_precision.py

```diff
@@ -0,0 +1,60 @@
+"""
+MAP metric.
+"""
+
+import torch
+
+
+def avg_precision(
+    outputs: torch.Tensor, targets: torch.Tensor, k=100
+) -> torch.Tensor:
+    """
+    Calculate the Average Precision (AP) for recsys
+    The precision metric summarizes the fraction of relevant items
+    out of the whole the recommendation list.
+    The average precision @ k (AP@k) metrics summarizes the average
```

Comment:
maybe, this could help a bit https://stackoverflow.com/questions/45534757/how-to-use-inline-code-with-a-trailing-whitespace